src/test/run-pass/backtrace.rs

```diff
@@ -59,6 +59,49 @@ fn runtest(me: &str) {
     assert!(s.contains("stack backtrace") && s.contains(&expected("foo")),
             "bad output: {}", s);
 
+    // Make sure than the short version cleans the backtrace.
+    let p = template(me).arg("fail").env("RUST_BACKTRACE", "1").spawn().unwrap();
+    let out = p.wait_with_output().unwrap();
+    assert!(!out.status.success());
+    let s = str::from_utf8(&out.stderr).unwrap();
+    let removed_symbols = &[
+        "std::sys::imp::backtrace",
+        "std::sys_common::backtrace",
+        "std::panicking",
+        "core::panicking",
+        "rust_begin_unwind",
+        "code::result::unwrap_failed",
+        "std::panicking::try::do_call",
+        "__rust_maybe_catch_panic",
+        "__libc_start_main",
+        "__rust_try",
+        "_start",
+    ];
+    for symbol in removed_symbols {
+        assert!(!s.contains(symbol),
+                "{} should be removed from the backtrace",
+                symbol);
+    }
+    assert!(s.contains(" 0:"), "the frame number should start at 0");
+
+    // Make sure than the short version cleans the backtrace.
+    let p = template(me).arg("fail").env("RUST_BACKTRACE", "full").spawn().unwrap();
+    let out = p.wait_with_output().unwrap();
+    assert!(!out.status.success());
+    let s = str::from_utf8(&out.stderr).unwrap();
+    let should_be_present = &[
+        "std::panicking",
+        "__rust_maybe_catch_panic",
+        "__libc_start_main",
```

Comment:
Asserting this symbol and `_start` below are present is likely to fail on non-linux platforms, perhaps those symbols could be tested on just Linux? Otherwise looks good to me!